File: iOS/DateMate/UI/Auth/ProfileSetup/ProfileSetupUiState.swift
//
//  ProfileSetupUiState.swift
//  DateMate
//
//  Created by 홍희표 on 2025/11/25.
//

import Foundation

struct ProfileSetupUiState {
    var name: String = ""
    var selectedImageUrls: [URL] = []
    var selectedGender: Gender? = nil
    var bio: String = ""
    var birthdayMillis: Int64? = nil
    var job: String = ""
    var isLoading: Bool = false
    var isSetupComplete: Bool = false
    var errorMessage: String? = nil
    var nameError: String? = nil
    var birthdayError: String? = nil
    var bioError: String? = nil
    var jobError: String? = nil
    var isSubmitEnabled: Bool {
        let isNameValid = !name.isEmpty && nameError == nil
        let isPhotoListValid = !selectedImageUrls.isEmpty
        let isGenderSelected = selectedGender != nil
        let isBirthdayValid = birthdayMillis != nil && birthdayError == nil
        let isBioValid = !bio.isEmpty && bioError == nil
        let isJobValid = !job.isEmpty && jobError == nil
        return isNameValid && isPhotoListValid && isGenderSelected &&
               isBirthdayValid && isBioValid && isJobValid
    }
}

File: Android/app/src/main/java/com/hhp227/datemate/ui/auth/profilesetup/ProfileSetupUiState.kt
package com.hhp227.datemate.ui.auth.profilesetup

import android.net.Uri
import com.hhp227.datemate.data.model.Gender

data class ProfileSetupUiState(
    val name: String = "",
    val selectedImageUris: List<Uri> = emptyList(),
    val selectedGender: Gender? = null,
    val bio: String = "",
    val birthdayMillis: Long? = null,
    val job: String = "",
    val isLoading: Boolean = false,
    val isSetupComplete: Boolean = false,
    val errorMessage: String? = null,
    val nameError: String? = null,
    val birthdayError: String? = null, // 생년월일 에러 추가
    val bioError: String? = null,   // 🆕 자기소개 오류 상태 추가
    val jobError: String? = null    // 🆕 직업 오류 상태 추가
) {
    val isSubmitEnabled: Boolean
        get() {
            val isNameValid = name.isNotBlank() && nameError == null
            val isPhotoListValid = selectedImageUris.isNotEmpty()
            val isGenderSelected = selectedGender != null
            val isBirthdayValid = birthdayError == null
            val isBioValid = bio.isNotBlank() && bioError == null && bio.length <= 200 // Bio 최대 길이 유효성 검사 추가 (200자)
            val isJobValid = job.isNotBlank() && jobError == null
            return isNameValid && isPhotoListValid && isGenderSelected && isBioValid && isBirthdayValid && isJobValid
        }
}